src/wp-includes/media.php

```diff
@@ -1989,6 +1989,17 @@ function wp_img_tag_add_srcset_and_sizes_attr( $image, $context, $attachment_id
  * @return string Converted `iframe` tag with `loading` attribute added.
  */
 function wp_iframe_tag_add_loading_attr( $iframe, $context ) {
+	// Iframes with fallback content (see `wp_filter_oembed_result()`) should not be lazy-loaded because they are
+	// visually hidden initially.
+	if ( false !== strpos( $iframe, ' data-secret="' ) ) {
+		return $iframe;
+	}
+
+	// Iframes should have source and dimension attributes for the `loading` attribute to be added.
+	if ( false === strpos( $iframe, ' src="' ) || false === strpos( $iframe, ' width="' ) || false === strpos( $iframe, ' height="' ) ) {
+		return $iframe;
+	}
```

Comment:
WP doesn't normalise either discovered or approved providers to use double quotes so I think this will need to account for both.

This test will fail:

```php
/**
 * @ticket 50756
 */
function test_wp_iframe_tag_add_loading_attr_single_quotes() {
	$iframe = "<iframe src='https://www.example.com' width='640' height='360'></iframe>";
	$iframe = wp_iframe_tag_add_loading_attr( $iframe, 'test' );

	$this->assertContains( ' loading="lazy"', $iframe );
}
```

Adding a single quote equivalent of `test_wp_iframe_tag_add_loading_attr_skip_wp_embed()` passes but I think that's luck rather than design.

Reply:
@peterwilsoncc That's a fair point, but I'd argue it's unrelated to this ticket and would increase the scope (the lines you're commenting on were just moved as is, they weren't changed). In the original tickets for image and iframe lazy-loading the topic of single quotes was explicitly discussed and decided as unnecessary to support.

We can certainly revisit that, but that should happen separately.